--- src/etcd/client.py~     2014-07-25 10:11:23.000000000 +0000
+++ src/etcd/client.py   2015-02-10 13:34:01.558814154 +0000
@@ -118,7 +118,10 @@
             # we need the set of servers in the cluster in order to try
             # reconnecting upon error.
             self._machines_cache = self.machines
-            self._machines_cache.remove(self._base_uri)
+            if self._base_uri in self._machines_cache:
+                self._machines_cache.remove(self._base_uri)
+            #else:
+            #    print "using proxy", self._base_uri  # XXX
         else:
             self._machines_cache = []
